contracts/pools/yv2/StableSwapYv2.vy

```diff
@@ -0,0 +1,996 @@
+future_fee: public(uint256)
+future_admin_fee: public(uint256)
+future_owner: public(address)
+
+is_killed: bool
+kill_deadline: uint256
+KILL_DEADLINE_DT: constant(uint256) = 2 * 30 * 86400
+
+@external
+def __init__(
+    _owner: address,
+    _coins: address[N_COINS],
+    _underlying_coins: address[N_COINS],
+    _pool_token: address,
+    _A: uint256,
+    _fee: uint256,
+    _admin_fee: uint256,
+):
+    """
+    @notice Contract constructor
+    @param _owner Contract owner address
+    @param _coins Addresses of ERC20 contracts of wrapped coins
+    @param _underlying_coins Addresses of ERC20 contracts of underlying coins
+    @param _pool_token Address of the token representing LP share
+    @param _A Amplification coefficient multiplied by n * (n - 1)
+    @param _fee Fee to charge for exchanges
+    @param _admin_fee Admin fee
+    """
+    for i in range(N_COINS):
+        assert _coins[i] != ZERO_ADDRESS
+        assert _underlying_coins[i] != ZERO_ADDRESS
+
+        # approve underlying coins for infinite transfers
+        _response: Bytes[32] = raw_call(
+            _underlying_coins[i],
+            concat(
+                method_id("approve(address,uint256)"),
+                convert(_coins[i], bytes32),
+                convert(MAX_UINT256, bytes32),
+            ),
+            max_outsize=32,
+        )
+        if len(_response) > 0:
+            assert convert(_response, bool)
+
+    self.coins = _coins
+    self.underlying_coins = _underlying_coins
+    self.initial_A = _A * A_PRECISION
+    self.future_A = _A * A_PRECISION
+    self.fee = _fee
+    self.admin_fee = _admin_fee
+    self.owner = _owner
+    self.kill_deadline = block.timestamp + KILL_DEADLINE_DT
+    self.lp_token = _pool_token
+
+
+@view
+@internal
+def _A() -> uint256:
+    """
+    Handle ramping A up or down
+    """
+    t1: uint256 = self.future_A_time
+    A1: uint256 = self.future_A
+
+    if block.timestamp < t1:
+        A0: uint256 = self.initial_A
+        t0: uint256 = self.initial_A_time
+        # Expressions in uint256 cannot have negative numbers, thus "if"
+        if A1 > A0:
+            return A0 + (A1 - A0) * (block.timestamp - t0) / (t1 - t0)
+        else:
+            return A0 - (A0 - A1) * (block.timestamp - t0) / (t1 - t0)
+
+    else:  # when t1 == 0 or block.timestamp >= t1
+        return A1
+
+
+@view
+@external
+def A() -> uint256:
+    return self._A() / A_PRECISION
+
+
+@view
+@external
+def A_precise() -> uint256:
+    return self._A()
+
+
+@view
+@internal
+def _stored_rates() -> uint256[N_COINS]:
+    # exchangeRateStored * (1 + supplyRatePerBlock * (getBlockNumber - accrualBlockNumber) / 1e18)
+    result: uint256[N_COINS] = PRECISION_MUL
+    for i in range(N_COINS):
+        rate: uint256 = cyToken(self.coins[i]).exchangeRateStored()
+        rate += rate * cyToken(self.coins[i]).supplyRatePerBlock() * (block.number - cyToken(self.coins[i]).accrualBlockNumber()) / PRECISION
```

Comment:
assigning `self.coins[i]` to a memory variable will reduce the number of storage loads here, and should help the bytecode size.